### DISCUSSION TITLE
Prevent creation of Fake(SPAM) user with Prestashop

### DISCUSSION THREAD

**Discussion by rbeau** [PrestaShop/PrestaShop#33217] [Q&A]
Hi everyone !

I'm writing here because I hadn't found any natural Prestashop solution to prevent SPAM registration. There a many registration every day.

![Screenshot-2023-07-11](https://github.com/PrestaShop/PrestaShop/assets/13245810/fad495b0-b5e9-4c60-b744-3884591df03d)

And those cause many problem, like Hotmail Staff (Abuse) team reported me many time this bug:

![Screenshot-2023-07-11-at-18-26-23-Message-d'origine](https://github.com/PrestaShop/PrestaShop/assets/13245810/665e88d5-6ec2-4cd1-a274-ad4e599e1628)

This is important avertissement, I must to find a solution or Web server will be blocked by Microsoft.

Could you please help me to find a real and natural Prestashop patch for this bug ?

Cordially,
Ridvan

**Comment by kpodemski** [comment]
You can try to use some of the third-party modules offered on PrestaShop marketplace or elsewhere, which will add some captcha preventing bot users from registering.

**Reply by rbeau** [reply]
I tried two separate modules: "Ever Ps Captcha" and "Custom Validate Anti Spam," but none of them solved the problem.

Prestashop must defend itself from spammers.

**Reply by kpodemski** [reply]
I agree! You [can help us](https://devdocs.prestashop-project.org/8/contribute/contribution-guidelines/) by submitting a Pull Request and contributing a solution you think it's best to fight the spam.

**Reply by MatShir** [reply]
We don't have the email validation after account creation

**Reply by matks** [reply]
Email validation will do not help here because the email addresses used by spam bots are valid 😉 . vofnomimib@outlook.com is valid.

**Reply by MatShir** [reply]
I mean validating a link on the email after submitting an account creation.

**Comment by matks** [comment]
> Prestashop must defend itself from spammers.

Interesting reading -> https://build.prestashop-project.org/news/2019/fighting-against-spamming-again/ 

(read "About CAPTCHAs" section at the end)

SPAM fighting is something that cannot happen in PrestaShop software because it is not designed to work this way. You see, PrestaShop is an on-premise open source software: you install it and it run on your server.

Known anti-spam systems (captcha) aim to filter bot users against human users. And this detection relies on complicated puzzle-solving that must change each time, and be constantly updated. Such a system can only be done efficiently by a [SaaS](https://en.wikipedia.org/wiki/Software_as_a_service) system, not a on-premise system.

If you search popular anti-spam systems you'll find they are all SaaS, not on-premise : reCAPTCHA, Cloudflare...

So solution is to use a module that uses an external SaaS system (free or on subscription). For examples https://addons.prestashop.com/fr/recherche?search_query=cloudflare (there might be others).
